dsd/constraints.py

```diff
@@ -677,20 +673,19 @@ def generate_sequence(self, rng: np.random.Generator, previous_sequence: Optiona
                 hamming_probabilities.append(val)
             # chooses random number of bases to be changed (steps)
             steps = np.random.choice(hamming_distances, p=hamming_probabilities) # seed?
-            adjacency_list = self.graph_hamming_distance(steps, previous_sequence)
-            while not adjacency_list[previous_sequence]:
-                # changes value of steps until such a neighbor of previous_sequence exists
+            adjacency_list = self.find_sequence_neighbors(steps, previous_sequence)
```

Comment:
Same here, call this variable `neighbors`. Although I suppose `neighbors` is not a great name either since they aren't neighbors in the Hamming graph unless `steps == 1`. Perhaps `steps_distance_sequences` is a better name for both variables.